runtime/stackmap/stackmap.c

```diff
@@ -350,7 +350,14 @@ mapStack(UDATA *scratch, UDATA totalStack, U_8 * map, J9ROMClass * romClass, J9R
 				}
 
 				if (pool[index].slot2) {
-					PUSH(OBJ);
+					/* Check for primitive values resolved from ConstantDynamic */
+					if (((pool[index].slot2 & J9DescriptionCpTypeMask) == J9DescriptionCpTypeConstantDynamic)
+					&& (pool[index].slot2 >> J9DescriptionReturnTypeShift)
```

Comment:
Do not use implicit zero comparisons. Also, strictly speaking the constant should be on the left of the comparison.

Reply:
fixed